python/lsst/obs/base/instrument.py

```diff
@@ -168,6 +169,31 @@ def fromName(cls, name, registry):
         instrument = doImport(cls)
         return instrument()
 
+    @staticmethod
+    def importAll(registry: Registry) -> None:
+        """Import all the instruments known to this registry.
+
+        This will ensure that all metadata translators have been registered.
+
+        Parameters
+        ----------
+        registry : `lsst.daf.butler.Registry`
+            Butler registry to query to find the information.
+
+        Notes
+        -----
+        It is allowed for a particular instrument class to fail on import.
+        This might simply indicate that a particular obs package has
+        not been setup.
+        """
+        dimensions = list(registry.queryDimensions("instrument"))
+        for dim in dimensions:
+            cls = dim.records["instrument"].class_name
+            try:
+                doImport(cls)
+            except Exception:
+                pass
```

Comment:
Do you want to log here in case the user's confused about what didn't get imported?

Reply:
I think that there might be many places where a failure here might be expected (such as the aforementioned failure to set up obs_decam) and people don't like irrelevant messages. If there is a real problem with the relevant instrument class we'll find out about it pretty quickly. I suppose I could be persuaded to use a debug log message here.

Reply:
Although there isn't a logger set up in that class at present so maybe I shouldn't do it just now. I have noticed though that the base class constructor takes arguments whereas I'm pretty sure we don't want Instrument classes to need arguments on construction.